packages/dart_locations_repository_interface/lib/src/locations_models.dart

```diff
@@ -1,2 +1,76 @@
+/// A single representation of an item at a location
+abstract interface class LocationItem {
+  /// The real-world location to which this item is bound
+  Location get location;
+
+  /// The identifier for this object, used to bind this to other representations
+  /// of the same object.
+  ///
+  /// Should at all times be unique within the context of this domain.
+  String get locationId;
+}
+
 /// A single entry for a location
-class LocationItem {}
+class DefaultLocationItem implements LocationItem {
+  /// Instantiates an item on a location
+  ///
+  /// The [locationId] and [location] are required to identify what and where.
+  ///
+  /// Additional parameters can be given, but are not required
+  DefaultLocationItem({
+    required this.location,
+    required this.locationId,
+    this.locationImageUrl,
+    this.locationDescription,
+    this.locationTitle,
+  });
+
+  @override
+  final Location location;
+  @override
+  final String locationId;
+
+  /// Optional image bound to this location
+  final String? locationImageUrl;
+
+  /// Optional title bound to this location
+  final String? locationTitle;
+
+  /// Optional description bound to this location
+  final String? locationDescription;
+}
+
+/// This defines the information needed to know where a physical location can
+/// be found in the world.
+class Location {
```

Comment:
Why would you not just use a factory constructor that normalizes it on construction, given that there is no usecase in representing long as anything other than a value in the range <-180,180>.

Reply:
I can imagine that as long as we are calculating, it can happen that locations exceed the normalized values. For example if a user scrolls around the world twice, some packages return the actual offset from the center rather than using the offset relative to the 0,0 coordinates. I think that having an explicit way to construct a normalized location is the best way to go. I added a Location.normalized() constructor